frontend/javascripts/maintenance_banner.tsx

```diff
@@ -0,0 +1,106 @@
+import {
+  MaintenanceInfo,
+  listCurrentAndUpcomingMaintenances,
+  updateNovelUserExperienceInfos,
+} from "admin/admin_rest_api";
+import { Alert } from "antd";
+import FormattedDate from "components/formatted_date";
+import { useInterval } from "libs/react_helpers";
+import { setActiveUserAction } from "oxalis/model/actions/user_actions";
+import { Store } from "oxalis/singletons";
+import React, { useState } from "react";
+
+export function MaintenanceBanner() {
+  const { activeUser } = Store.getState();
+  const [currentAndUpcomingMaintenances, setcurrentAndUpcomingMaintenances] = useState<
+    Array<MaintenanceInfo>
+  >([]);
+  const [position, setPosition] = useState<Object>({ top: 48 });
+  const [isTop, setIsTop] = useState(true);
+  useInterval(
+    async () => {
+      if (activeUser == null) return;
```

Comment:
wouldn't maintenances also be relevant for logged out users that view public datasets? you probably did that because you cannot store the `latestAcknowledgedMaintenanceInfo` for logged-out users, right? however, for running maintenances, this property isn't used anyway. I'd suggest to show running maintenances for all users and upcoming maintenances only for logged-in users.

Reply:
actually I just thought that this info wouldnt be important for logged-out users, but what I didnt have in mind is that they could actively be viewing datasets, so I think you're right that this info matters for everyone